module3/lessons/intro-to-cypress-testing.md

```diff
@@ -0,0 +1,633 @@
+<section class='dropdown'>
+### Possible Solution
+
+**(hint: you only need to run a single spec file at a time)**
+- We need to have our login on the dashboard within our beforeEach, since we are starting a new spec file, and our dashboard spec is dependent on a user being logged in.
+- Our dashboard spec is dependent on our feedback request,teammates request, the user's data and their teammates data so we need to stub those network requests as well.
+- 
+
+```js
+// dashboard_spec.js
+describe('Dashboard view', () => {
+    beforeEach(() => {
+      // Login the user and stub the network requests 
+      cy.intercept('POST', 'http://localhost:3001/api/v1/login', {
+        statusCode: 201,
+        body : {
+          id:2,
+          image: "https://ca.slack-edge.com/T029P2S9M-U37MJAV0T-007ccf2f5eb2-512",
+          email: 'leta@turing.io',
+          password: 'keane20'
+        }
+      })
+      // Stub the network requests for the user's teammates
+      cy.intercept(`http://localhost:3001/api/v1/users/2/teammates`, {
+      "teammates": [
+        {
+          email: "hannah@turing.io",
+          id: 1,
+          image: "https://ca.slack-edge.com/T029P2S9M-UPE0QSWEQ-d4bebe6f4d88-512",
+          name: "Hannah Hudson",
+          delivered: false
+        },
+        {
+          email: "khalid@turing.io",
+          id: 3,
+          image: "https://ca.slack-edge.com/T029P2S9M-UDR1EJKFS-9351230a5443-512",
+          name: "Khalid Williams",
+          delivered: true
+        }
+      ]
+    });
+    // Stub the network requests for the user's feedback
+    cy.intercept(`http://localhost:3001/api/v1/users/2/feedback`, {
+      feedback: [
+        {
+          feedback: "Your feedback game is TOO strong.",
+          senderId: 4,
+          receiverId: 2
+        },
+        {
+          feedback: "I appreciate your positive energy and how hard you work in supporting both students and other instructors alike.",
+          senderId: 11,
+          receiverId: 2
+        }
+      ]
+    });
+    // Stub the network requests for the user's data
+     cy.intercept("GET", "http://localhost:3001/api/v1/users/4", {
+      id: 4,
+      name: "Scott Ertmer",
+      image: "https://ca.slack-edge.com/T029P2S9M-UJ910QEJF-7244f37f7e12-512",
+      email: "scott@turing.io",
+      password: "ertmer20",
+    });
+    cy.intercept("GET", "http://localhost:3001/api/v1/users/11", {
+      id: 11,
+      name: "Travis Rollins",
+      image: "https://ca.slack-edge.com/T029P2S9M-U4R41TZD2-7661f06e8c71-512",
+      email: "travis@turing.io",
+      password: "rollins20",
+    });
+    cy.visit('http://localhost:3000/')
+  })
+})
+```
+**As you see our beforeEach is doing a lot of things, this is a good example of where we would refactor our code to make our tests more performant.**
+- We can create fixtures to help us with this.
```

Comment:
Can we add one more section here where we actually walk them through using a fixture for at least one of the intercepts above?   Maybe also an example that uses the 'body' property?

Reply:
Thank you @hfaerber, I've added 2 more sections for using fixture and body with intercept. You can find them [here](https://github.com/turingschool/curriculum-site/pull/274/commits/e2ba3e59c79370948eca9588ac0444af5fc4ac02)